### PR TITLE
feat: add file upload hook

### PR DESCRIPTION
**Pull request by SamCreasey**
Linked Pr: https://github.com/Frontify/clarify/pull/10315

### CODE REVIEW

File: src/react/useFileUpload.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        resetState();
+        const fileArray = getFilesAsArray(files);
+
+        if (fileArray.length === 0) {
+            return;
+        }
+
+        const message = {
+            files: fileArray,
+            formData: projectId ? [{ project_id: projectId }] : null,
```
Comment (SamCreasey) [inline src/react/useFileUpload.ts:131]: Can the projectID be omitted completely from the files now with the update?

File: src/react/useFileUpload.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+import { useEffect, useRef, useState } from 'react';
```
Comment (SamuelAlev) [inline src/react/useFileUpload.ts:2]:
```suggestion
/* (c) Copyright Frontify Ltd., all rights reserved. */

import { useEffect, useRef, useState } from 'react';
```

File: src/react/useFileUpload.ts (hunk 1 above)
Comment (SamuelAlev) [reply src/react/useFileUpload.ts:131]: Yup

File: src/workers/upload.worker.js [hunk 3]
```diff
@@ -209,7 +209,7 @@ function error(index, file, fileMetadata) {
     const error = fileMetadata?.error ?? 'Unable to process file.';
 
     self.postMessage({
-        event: 'onAssetFail',
```
Comment (SamuelAlev) [inline src/workers/upload.worker.js:212]: `onAssetFail` is used in clarify code, make sure to also change it there when updating app-bridge

File: src/react/useFileUpload.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+export const useFileUpload = (props?: UseFileUploadParameters): UseFileUploadReturnTypes => {
+    const results = useRef<FrontifyFile[]>([]);
+
+    const [doneAll, setDoneAll] = useState(false);
+
+    const { onUploadProgress, onUploadProgressAll, onUploadDone, onUploadDoneAll, onUploadFail, onUploadFileFail } =
+        props ?? {};
+
+    const projectId = window.application?.sandbox?.config?.context?.project?.id;
```
Comment (SamuelAlev) [inline src/react/useFileUpload.ts:37]: Can be removed

File: src/react/useFileUpload.ts (hunk 1 above)
Comment (SamCreasey) [reply src/react/useFileUpload.ts]: removed

File: src/react/useFileUpload.ts (hunk 2 above)
Comment (SamCreasey) [reply src/react/useFileUpload.ts]: fixed

File: src/react/useFileUpload.ts (hunk 4 above)
Comment (SamCreasey) [reply src/react/useFileUpload.ts]: fixed

File: src/workers/upload.worker.js (hunk 3 above)
Comment (SamCreasey) [reply src/workers/upload.worker.js:212]: will check it out

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**